packages/fxa-content-server/app/scripts/views/settings/recovery_codes.js

```diff
@@ -79,9 +79,11 @@ const View = FormView.extend({
     const input = this.getElementValue('input.recovery-code');
     const codes = this.model.get('recoveryCodes');
     if (!codes.includes(input.toLowerCase())) {
+      const e =
+        input.length > 0 ? 'INVALID_RECOVERY_CODE' : 'RECOVERY_CODE_REQUIRED';
```

Comment:
Would you be able to check the validation of the input directly? Kind of like https://github.com/mozilla/fxa/blob/d667541a9e2568e331fb348f0997974a201895ac/packages/fxa-content-server/app/scripts/views/form.js#L275.

Since the input has the class `recovery-code` I would think rules in https://github.com/mozilla/fxa/blob/8701348cdd79dbdc9879b2b4a55a23a135a32bc1/packages/fxa-content-server/app/scripts/views/elements/recovery-code-input.js#L26 would still apply?

Reply:
thanks, i saw that but decided against it because it'd split the checking into two places. The static validation in `validate.js` isn't enough. If anything I'd prefer to remove the element validation instead